### DISCUSSION TITLE
FiniteStrainSingleCrystal brick and user defined SlidingSystems

### DISCUSSION THREAD

**Discussion by SCrenSC** [thelfer/tfel#407] [Q&A] [answered]
Hi,
could we use FiniteStrainSingleCrystal brick with sliding systems and interaction matrix both imported from Mfront files? 
I ask this because I already have the interaction matrix and slip systems as 2 separate Mfront files. 

1. I tested that I could keep the @SlidingSystems command with imported interactionmatrix.mfront, but the slip plane/directions' order differs from my existing interaction matrix file. (Change the interaction matrix file order is ok but just wondering if it is possible to use the same.)

2. If I use @InteractionMatrix together with @SlidingSystems, it asks me to define 28 parameters to build the interaction matrix. I feel it is probably simpler to just use my previous files.

3. However, when I deleted the @SlidingSystem command I have the following error.
FiniteStrainSingleCrystalBrick::completeVariableDeclaration: no slip systems defined

Thank you for your help.

**Comment by thelfer** [comment, marked as answer]
Dear @SCrenSC,
Unfortunately, you can't. Sorry for that, but the brick can't work with  your external file.
Regards,
Thomas
